### DISCUSSION TITLE
object count and size of 5m resolution metrics greater than raw

### DISCUSSION THREAD

**Discussion by grimz-ly** [thanos-io/thanos#6914] [General]
why do thanos 5m resolution metrics have more objects than raw resolution metrics for downsampled blocks? I may be missing something important below, but samples and chucks get reduced. the 5m resolution just includes a lot more objects. 214 5m chunks vs 118 raw

> docker-compose exec mc mc --insecure du thanos/ruler-s3/01HFF7GR8KMS8DVPC0MSDTM5KF
68GiB   120 objects     ruler-s3/01HFF7GR8KMS8DVPC0MSDTM5KF

Series: 17,777,516
Samples: 35,664,612,134
Chunks: 869,334,862

> docker-compose exec mc mc --insecure du thanos/ruler-s3/01HFG1VGSWYCZ9FX07NKGY00AC
110GiB  216 objects     ruler-s3/01HFG1VGSWYCZ9FX07NKGY00AC

Series: 17,777,516
Samples: 20,206,546,006
Chunks: 63,754,482

> docker-compose exec mc mc --insecure du thanos/rruler-s3/01HFG8YWTRWMA1RWKHP3H6NGAV
17GiB   30 objects      ruler-s3/01HFG8YWTRWMA1RWKHP3H6NGAV

Series: 17,777,516
Samples: 1,720,580,173
Chunks: 27,751,365

**Comment by yeya24** [comment]
> I may be missing something important below, but samples and chucks get reduced. the 5m resolution just includes a lot more objects. 214 5m chunks vs 118 raw

Based on the command you ran, `01HFF7GR8KMS8DVPC0MSDTM5KF` is the raw block and `01HFG1VGSWYCZ9FX07NKGY00AC` is the 5m block if I understand correctly.

Regarding the point of number of objects for 5m and raw blocks, I think it is expected. The chunk in 5m block is of AggrChunk type which contains 5 different types of aggregations, each aggregation is a XOR type chunk. While a chunk in raw block is just one XOR type chunk.

So it is possible that the chunk in 5m block is larger than chunk from raw block, depending on your scrape interval. If you have a very small scrape interval like 5s, I guess having 5m downsampled block helps reduce your block size. But if your scrape interval is close to 5m then it is expected to see your block is bigger.
